src/scrabble.js

```diff
@@ -1,5 +1,172 @@
+      '[',
+      ']',
+      ' ',
+      '',
+      '.'
+    ]
+    return acceptableChars.filter((pass) => pass.includes(charExtract))
+  }
+  for (let i = 0; i < caseSensitivity.length; i++) {
+    charExtract = caseSensitivity.charAt(i)
+    if (
+      charExtract === 'A' ||
+      charExtract === 'I' ||
+      charExtract === 'E' ||
+      charExtract === 'O' ||
+      charExtract === 'U' ||
+      charExtract === 'L' ||
+      charExtract === 'N' ||
+      charExtract === 'R' ||
+      charExtract === 'S' ||
+      charExtract === 'T'
+    ) {
+      charPoints = 1
+    }
+
+    if (charExtract === 'D' || charExtract === 'G') {
+      charPoints = 2
+    }
+
+    if (
+      charExtract === 'B' ||
+      charExtract === 'C' ||
+      charExtract === 'M' ||
+      charExtract === 'P'
+    ) {
+      charPoints = 3
+    }
+
+    if (
+      charExtract === 'F' ||
+      charExtract === 'H' ||
+      charExtract === 'V' ||
+      charExtract === 'W' ||
+      charExtract === 'Y'
+    ) {
+      charPoints = 4
+    }
+
+    if (charExtract === 'K') {
+      charPoints = 5
+    }
+
+    if (charExtract === 'J' || charExtract === 'X') {
+      charPoints = 8
+    }
+
+    if (charExtract === 'Q' || charExtract === 'Z') {
+      charPoints = 10
+    }
+
+    if (
+      charExtract === ' ' ||
+      charExtract === '' ||
+      charExtract === '{' ||
+      charExtract === '}' ||
+      charExtract === '[' ||
+      charExtract === ']' ||
+      charExtract === '.'
+    ) {
+      charPoints = 0
+    }
+
+    if (caseSensitivity.includes(`{${charExtract}}`)) {
+      charPoints *= 2
+    }
+
+    if (caseSensitivity.includes(`[${charExtract}]`)) {
+      charPoints *= 3
+    }
+
+    if (
+      caseSensitivity.search('{') === 0 &&
+      caseSensitivity.search('}') === caseSensitivity.length - 1
+    ) {
+      charPoints *= 2
+    }
+
+    if (
+      caseSensitivity.search(/\[/) === 0 &&
+      caseSensitivity.search(/\]/) === caseSensitivity.length - 1
+    ) {
+      charPoints *= 3
+    }
+
+    if (charExtract === '{' || charExtract === '}') {
+      doublePoints += 1
+    }
+
+    if (charExtract === '[' || charExtract === ']') {
+      triplePoints += 1
+    }
+
+    if (!incorrectTokenControl().includes(charExtract)) {
+      return 0
+    }
+
+    scrabbleScore += charPoints
+  }
+
+  if (doublePoints === 1 || triplePoints === 1) {
+    return 0
+  } else if (
+    caseSensitivity.search(/\[/) === 0 &&
+    caseSensitivity.search('{') === 1 &&
+    caseSensitivity.search('}') === caseSensitivity.length - 2 &&
+    caseSensitivity.search(/\]/) === caseSensitivity.length - 1
+  ) {
+    return scrabbleScore * 2
+  } else if (
+    caseSensitivity.search('{') === 0 &&
+    caseSensitivity.search(/\[/) === 1 &&
+    caseSensitivity.search(/\]/) === caseSensitivity.length - 2 &&
+    caseSensitivity.search('}') === caseSensitivity.length - 1
+  ) {
+    return scrabbleScore * 3
+  }
+
+  return scrabbleScore
 }
 
 module.exports = scrabble
```

Comment:
One thing I'm not quite picking up when reading your code is how exactly it's checking for the following:

1) Cases where it's a double and triple word score at the same time.
2) How it's checking for edge cases such as {d}o{g}

Overall I think that breaking your code up a bit more in to different functions would make it easier to follow. Good job though.

Reply:
![Screenshot 2023-03-17 144948](https://user-images.githubusercontent.com/124312512/225923848-43c9c327-463a-4b37-8418-ceb281a5cfb2.png)
Hi Alex, so for the cases where it's a double or triple word, the picture shows the code that executes that. So if you look in the condition of the if statement, I use .search() to check if one of [ or { is at the beginning of each string. I thought it would always be in the indexed position of 0 or 1 regardless of the argument input in the function. Then to seek for ] or } I figured it would always be at the end of whatever caseSensitivity.length() - 1, and this ensures that regardless of the argument, the indexed position would always be found using .length(). 

![image](https://user-images.githubusercontent.com/124312512/225925752-36655b62-68ed-4046-bd3b-651d0c964b12.png)
So for the edge cases, this is the code that checks for that. So since I put this in the loop, each time the loop iterates and it extracts a character from the argument and it passes it in the if statement condition if (caseSensitivity.includes(`{${charExtract}}`)). This checks the entire string if the current character that was extracted in the loop is inside  {} or [] regardless of how many times they appear. i.e ({d}{o}{g} or {d}o{g} or d{o}g)

If you have any more questions write to me!
I hope I explained myself well enough and also I hope I understood and answered your questions clearly. Let me know!

Reply:
![Screenshot 2023-03-17 144948](https://user-images.githubusercontent.com/124312512/225927428-19cac027-506c-402e-853b-64ad8ae1748d.png)
Hi Alex, so for the cases where it's a double or triple word, the picture shows the code that executes that. So if you look in the condition of the if statement, I use .search() to check if one of [ or { is at the beginning of each string. I thought it would always be in the indexed position of 0 or 1 regardless of the argument input in the function. Then to seek for ] or } I figured it would always be at the end of whatever caseSensitivity.length() - 1, and this ensures that regardless of the argument, the indexed position would always be found using .length().

![Screenshot 2023-03-17 145721](https://user-images.githubusercontent.com/124312512/225927492-6d3c2eba-d3cd-45c1-9e78-f8bf25c73246.png)
So for the edge cases, this is the code that checks for that. So since I put this in the loop, each time the loop iterates and it extracts a character from the argument and it passes it in the if statement condition if (caseSensitivity.includes({${charExtract}})). This checks the entire string if the current character that was extracted in the loop is inside {} or [] regardless of how many times they appear. i.e ({d}{o}{g} or {d}o{g} or d{o}g)

If you have any more questions write to me!
I hope I explained myself well enough and also I hope I understood and answered your questions clearly. Let me know!